src/rgw/rgw_threadpool.h

```diff
@@ -0,0 +1,121 @@
+#ifndef _RGW_THREADPOOL_
+#define _RGW_THREADPOOL_
+
+#include <queue>
+
+class WorkerThread {
```

Comment:
other code in rgw sends http requests with `RGWHTTPClient` and `RGWHTTPManager`, which uses the `curl_multi` interface to do this with a single thread. we'll want to use that instead